pkg/health/health.go

```diff
@@ -28,24 +30,33 @@ type Result struct {
 	Output  string
 }
 
-func (h ConsulHealthChecker) Service(serviceID string) (map[string][]Result, error) {
-	entries, _, err := h.client.Health().Service(serviceID, "", false, nil)
+func (h ConsulHealthChecker) Service(serviceID string) (map[string][][]Result, error) {
+	catalogEntries, _, err := h.client.Health().Service(serviceID, "", false, nil)
 	if err != nil {
 		return nil, util.Errorf("/health/service failed for %q: %s", serviceID, err)
 	}
 
-	ret := make(map[string][]Result)
-	for _, entry := range entries {
-		ret[entry.Node.Node] = make([]Result, 0, len(entry.Checks))
+	ret := make(map[string][][]Result)
+	for _, entry := range catalogEntries {
+		// get entry in KV store for this service/node
+		kvEntry, err := h.consulStore.GetHealth(serviceID, entry.Node.Node)
+		if err != nil {
+			return nil, util.Errorf("/health/service failed for %q on node %s: %s", serviceID, entry.Node.Node, err)
+		}
+
+		res := make([]Result, 0, len(entry.Checks))
 		for _, check := range entry.Checks {
-			ret[entry.Node.Node] = append(ret[entry.Node.Node], consulCheckToResult(*check))
+			res = append(res, consulCheckToResult(*check))
 		}
+		kvRes := []Result{consulWatchToResult(kvEntry)}
+		// place worst result from catalog and result from KV store in map
+		ret[entry.Node.Node] = [][]Result{kvRes, res}
```

Comment:
Rather than return a single list of results it now returns one list per source of health info. I think this would be confusing for the client as is. The other options were to append kv store results to the catalog results and the api would be the same as it was before. However if this is done there is no way to distinguish between the two sources of health.

I think the best way would be to return FindWorst(all results) and return a single result. This would result in changes to the API, but unless there is a strong reason for wanting to return all the results this option might be the most clear. All calls to Service and WatchNodeService call FindWorst on the results anyway. The downside of this means we're deciding for the client what 'worst' is.